src/main/java/seedu/algobase/logic/commands/EditTagCommand.java

```diff
@@ -0,0 +1,105 @@
+package seedu.algobase.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.algobase.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.algobase.model.Model.PREDICATE_SHOW_ALL_TAGS;
+
+import java.util.List;
+import java.util.Optional;
+
+import seedu.algobase.commons.core.Messages;
+import seedu.algobase.commons.core.index.Index;
+import seedu.algobase.logic.commands.exceptions.CommandException;
+import seedu.algobase.model.Model;
+import seedu.algobase.model.tag.Tag;
+import seedu.algobase.model.tag.exceptions.TagNotFoundException;
+
+/**
+ * Edits the details of an existing Tag in the algobase.
+ */
+public class EditTagCommand extends Command {
+
+    public static final String COMMAND_WORD = "edittag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the Tag identified "
+            + "by the index number used in the displayed Tag list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_TAG + "TAG] "
+            + "Example: " + COMMAND_WORD + " 1 " + PREFIX_TAG + "Easy";
+    public static final String MESSAGE_EDIT_PROBLEM_SUCCESS = "Edited Tag: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_TAG = "This Tag already exists in the algobase.";
+
+    private final Index index;
+    private final Optional<String> name;
```

Comment:
Can you justify the usage of `Optional` here? In `EditCommand` it is used because not all fields need to be provided, however here `name` is required right?